docs/docs/learn/shards/working-with-data.md

```diff
@@ -1,11 +1,236 @@
+    (defpure wire-name
+      ;; your shards here
+      )
+    ```
+
+In the example below, you can see how using `Step` on a Pure Wire still does not give it access to the parent Wire's variables.
+
+=== "Command"
+    ```{.clojure .annotate linenums="1"}
+    (defmesh main)
+
+    (defwire unpure-wire
+      (Get .x :Default 0) (Log))
+
+    (defpure pure-wire
+      (Get .x :Default 0) (Log))
+
+    (defwire main-wire
+      5 >= .x
+      (Step unpure-wire)
+      (Step pure-wire))
+
+    (schedule main main-wire)
+    (run main)
+    ```
+
+=== "Output"
+    ```
+    [unpure-wire] 5
+    [pure-wire] 0
+    ```
+
+### Defined Constants
+
+If you define a constant in your program, it will have a global scope and can be accessed by any Wire in your program.
+
+=== "Command"
+    ```{.clojure .annotate linenums="1"}
+    (defmesh main)
+
+    (def x 1)
+
+    (defwire get-x
+      x (Log "x"))
+
+    (schedule main get-x)
+    (run main)
+    ```
+
+=== "Output"
+    ```{.clojure .annotate linenums="1"}
+    [get-x] x: 1
+    ```
+
+!!! note
+    Note that the constant `x` defined in the example above is named differently from variables, which would have been `.x` in this case.
+
+## Passthrough
+
+Passthrough determines if data can pass through a Wire unaltered. It allows you to better control the state of the data moving through your program.
+
+Wires will have passthrough disabled by default. In order to enable passthrough for Wires, we employ the shard [`Sub`](../../../reference/shards/General/Sub/). Any shards passed into the `Shards` parameter of `Sub` will run as per usual, except that the final output will be replaced with the initial input passed into `Sub`, thereby creating a passthrough effect.
+
+`Sub` has an alias `|` which eliminates the need for `->` to group shards together when passed into its `Shards` parameter.
+
+=== "Sub Example"
+    ```{.clojure .annotate linenums="1"}
+    (defmesh main)
+
+    (defwire sub-test
+      1 >= .x (Log "Before Sub")
+      (Sub
+       :Shards
+       (-> (Math.Add 2) > .x
+           .x (Log "In Sub")))
+      (Log "After Sub"))
+
+    (schedule main sub-test)
+    (run main)
+    ```
+
+=== "Sub Example with |"
+    ```{.clojure .annotate linenums="1"}
+    (defmesh main)
+
+    (defwire sub-test
+      1 >= .x (Log "Before Sub")
+      (|(Math.Add 2) > .x
+        .x (Log "In Sub")) 
+      (Log "After Sub"))
+
+    (schedule main sub-test)
+    (run main)
+    ```
+  
+=== "Output"
+    ```
+    [sub-test] Before Sub: 1
+    [sub-test] In Sub: 3
+    [sub-test] After Sub: 1
+    ```
+
+In the example below, John attempts to find out the price of an apple in different currencies. The base price of 1 USD is passed into the Wire, and goes through a series of shards that each performs mathematical operations on it to obtain its foreign value.
+
+In order to keep the initial value unchanged as it passes through Wires, passthrough has to be enabled for each Wire.
```

Comment:
Need to see the corresponding code.

Reply:
I think I will leave the code out for this example. The actual code is a little too verbose and might not make much sense (especially since a string will be created for logging, so the actual output without placing the converted value at the end would be a string instead of a float.